File: backend/test_upload.js
const axios = require('axios');
const FormData = require('form-data');
const fs = require('fs');

async function testUpload() {
    try {
        // 1. Login
        console.log('Login yapılıyor...');
        const loginResponse = await axios.post('http://localhost:3000/api/users/login', {
            eposta: 'u@gmail.com',
            sifre: '1234'
        });
        
        console.log('Login başarılı!');
        console.log('User ID:', loginResponse.data.data.id);
        const token = loginResponse.data.data.token;
        console.log('Token:', token.substring(0, 50) + '...');
        
        // 2. Upload Photo
        console.log('\nFotoğraf yükleniyor...');
        const form = new FormData();
        form.append('photo', fs.createReadStream('images.jpg'));
        
        const uploadResponse = await axios.post('http://localhost:3000/api/upload/photo', form, {
            headers: {
                ...form.getHeaders(),
                'Authorization': `Bearer ${token}`
            }
        });
        
        console.log('\nFotoğraf yükleme başarılı!');
        console.log('Response:', JSON.stringify(uploadResponse.data, null, 2));
        
    } catch (error) {
        console.error('Hata:', error.response ? error.response.data : error.message);
    }
}

testUpload();

File: backend/test_upload.py
import requests
import json

# 1. Login
login_url = 'http://localhost:3000/api/users/login'
login_data = {'eposta': 'u@gmail.com', 'sifre': '1234'}
login_response = requests.post(login_url, json=login_data)
login_result = login_response.json()

print(f"Login Status: {login_response.status_code}")
print(f"Login Response: {json.dumps(login_result, indent=2)}")

if login_result.get('success'):
    token = login_result['data']['token']
    print(f"\nToken: {token[:50]}...")
    
    # 2. Upload Photo
    upload_url = 'http://localhost:3000/api/upload/photo'
    headers = {'Authorization': f'Bearer {token}'}
    
    with open('images.jpg', 'rb') as f:
        files = {'photo': ('images.jpg', f, 'image/jpeg')}
        upload_response = requests.post(upload_url, headers=headers, files=files)
    
    print(f"\nUpload Status: {upload_response.status_code}")
    print(f"Upload Response: {json.dumps(upload_response.json(), indent=2)}")
else:
    print("Login failed!")
